src/module.c

```diff
@@ -8734,6 +8848,14 @@ int RM_ACLCheckCommandPermissions(RedisModuleUser *user, RedisModuleString **arg
     return REDISMODULE_OK;
 }
 
+/* Same as RedisModule_ACLCheckCommandPermissions, but infers the current user from the RedisModuleCtx */
+int RM_ACLCheckCommandPermissionsCurrentUser(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
```

Comment:
i guess this API isn't really necessary, right?
it can be done with a combination of RM_GetCurrentUserName, RM_GetModuleUserFromUserName, and RM_ACLCheckCommandPermissions.
despite being cumbersome, i think there's an advantage with working with smaller building blocks that can achieve more rather than an API for each specific purpose.

Reply:
yes, except a comment we had that this was a lot of allocations for each call when in practice we didn't need to to do any of them and that was "bad".  @tezc want to chime in?

Reply:
Yes, it was just to avoid 2 unnecessary allocations + a lookup. Let's have a benchmark, see if we really need this. Maybe we can consider something in the middle, introducing a bit more generic `RM_GetCurrentUser()` (single allocation) and then calling `RM_ACLCheckCommandPermissions()`.

Reply:
removed this code for now.  can put it back if needed, less needed if we have dry run (and made more efficient)